storage/src/main/java/zipkin2/storage/kafka/streams/TraceStoreSupplier.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright 2019 jeqo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.kafka.streams;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.streams.state.WindowStore;
+import org.apache.kafka.streams.state.WindowStoreIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin2.DependencyLink;
+import zipkin2.Span;
+import zipkin2.storage.kafka.streams.serdes.DependencyLinkSerde;
+import zipkin2.storage.kafka.streams.serdes.NamesSerde;
+import zipkin2.storage.kafka.streams.serdes.SpanIdsSerde;
+import zipkin2.storage.kafka.streams.serdes.SpansSerde;
+
+/**
+ * Aggregation and storage of spans into traces.
+ */
+public class TraceStoreSupplier implements Supplier<Topology> {
```

Comment:
this looks like the more interesting file :)

Reply:
it does :D